src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -778,6 +778,48 @@ public async Task GoAwayFrame_UnprocessedStreamFirstRequestWaitsUntilSecondFinis
             }
         }
 
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task GoAwayFrame_RequestInFlight_Finished()
+        {
+            using (await Watchdog.CreateAsync())
+            using (var server = Http2LoopbackServer.CreateServer())
+            using (HttpClient client = CreateHttpClient())
+            {
+                Task<HttpResponseMessage> sendTask = client.GetAsync(server.Address);
+                Http2LoopbackConnection connection = await server.EstablishConnectionAsync();
```

Comment:
We generally prefer the CreateClientAndServerAsync helper, as it not only provides structure, it also includes a built-in timeout, avoids issues that could occur in a case like this if the client failed to connect to the server and then the server just waits forever for a connection that'll never happen, etc.